### PR TITLE
Node bootstrapping for all new nodes

### PR DESCRIPTION
**Pull request by rjshrjndrn**
Adding role to install, node-exporter, default ulimits and optional azure cli installation.

### Type of change

Please choose appropriate options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Enhancement
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes in the below checkboxes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Ran Test A
- [ ] Ran Test B

**Test Configuration**:
* Software versions:
* Hardware versions:

### Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by keshavprasadms** [comment]
Do you think we can add process exporter also as part of bootstrap? We can move that prometheus / scraping to kubernetes prom?

**Comment by rjshrjndrn** [comment]
> Do you think we can add process exporter also as part of bootstrap? We can move that prometheus / scraping to kubernetes prom?

I don't think so, as process exporter is specific to process. This is going to run when you create a bare machine.

File: ansible/bootstrap.yml [hunk 1]
```diff
@@ -1,37 +1,43 @@
-- hosts: 'all'
+- hosts: "{{ hosts| default("all") }}"
```
Comment (keshavprasadms) [inline ansible/bootstrap.yml:1]: This is a syntax error. You cannot have double quotes in double quotes.

**Review by keshavprasadms — CHANGES_REQUESTED** [review]
Nice work. Lets fix some code.

**Comment by keshavprasadms** [comment]
> > Do you think we can add process exporter also as part of bootstrap? We can move that prometheus / scraping to kubernetes prom?
> 
> I don't think so, as process exporter is specific to process. This is going to run when you create a bare machine.

Not really, we can monitor all the process also. Its not specific to a process.

File: ansible/bootstrap.yml (hunk 1 above)
Comment (keshavprasadms) [reply ansible/bootstrap.yml]: As a collaborator on the repo, I have fixed this.

**Review by keshavprasadms — APPROVED** [review]

**Pull request merged**